src/twp-editor-plugin/index.ts

```diff
@@ -53,6 +57,104 @@ export function twpEditorPlugin(options: TwpEditorPluginOptions): Rule {
       rules.push(mergeWith(pluginStateTemplateSource, MergeStrategy.Error));
     }
 
+    rules.push(exportPluginFromIndex(name));
+    rules.push(importPluginToCreatePluginsList(name));
+
     return chain(rules)(tree, context);
   };
 }
+
+function exportPluginFromIndex(pluginName: string): Rule {
+  return (tree: Tree) => {
+    const path = `${pluginBasePath}/index.ts`;
+    const buffer = tree.read(path);
+    if (!buffer) {
+      throw new SchematicsException(`File ${path} not found`);
+    }
+    const recorder = tree.beginUpdate(path);
+    recorder.insertRight(
+      buffer.length,
+      `export { default as ${strings.camelize(
+        pluginName
+      )}Plugin } from './${strings.dasherize(pluginName)}';\n`
+    );
+    tree.commitUpdate(recorder);
+    return tree;
+  };
+}
+
+function importPluginToCreatePluginsList(pluginName: string): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const path = `${createEditorPath}/create-plugins-list.ts`;
+    const buffer = tree.read(path);
+    if (!buffer) {
+      throw new SchematicsException(`File ${path} not found`);
+    }
+
+    const recorder = tree.beginUpdate(path);
+    const source = ts.createSourceFile(
+      path,
+      buffer.toString(),
+      ts.ScriptTarget.Latest,
+      true
+    );
+    const nodes = getSourceNodes(source);
+
+    const importClosingBrace = nodes
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.ImportDeclaration)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.ImportClause)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.NamedImports)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.CloseBraceToken);
+    if (!importClosingBrace) {
+      throw new SchematicsException(`Error locating imports in ${path}`);
+    }
+
+    recorder.insertLeft(
+      importClosingBrace.pos,
+      `\n  ${strings.camelize(pluginName)}Plugin,`
+    );
+
+    const pluginReturnStatement = nodes
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.FunctionDeclaration)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.Block)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.SyntaxList)
+      ?.getChildren()
+      .find((node: ts.Node) => node.kind === ts.SyntaxKind.ReturnStatement);
+    if (!pluginReturnStatement) {
+      throw new SchematicsException(
+        `Error locating plugin return statement in ${path}`
+      );
+    }
+
+    recorder.insertLeft(
+      pluginReturnStatement.pos,
```

Comment:
`insertLeft` inserts before content I imagine? So should we add new lines afterwards instead of before then? I think will look like:
```
plugins.push(somePlugin()); return plugins;
```

Reply:
I think `insertLeft` automatically moves the cursor to the end of the previous block. I did have it adding new lines after initially but it ended up looking like this:
```
const plugins = getDefaultPluginsList(props);

if (props.allowExisting) {
plugins.push(existingPlugin());
}plugins.push(mySamplePlugin());



return plugins;
```